iotdb-core/consensus/src/main/java/org/apache/iotdb/consensus/ratis/RatisConsensus.java

```diff
@@ -864,14 +868,30 @@ private RatisClient getRaftClient(RaftGroup group) throws ClientManagerException
     }
   }
 
+  private RatisClient getConfigurationRaftClient(RaftGroup group) throws ClientManagerException {
+    try {
+      return reconfigurationClientManager.borrowClient(group);
+    } catch (ClientManagerException e) {
+      logger.error("Borrow client from pool for group {} failed.", group, e);
+      // rethrow the exception
+      throw e;
+    }
+  }
+
   private RaftClientReply sendReconfiguration(RaftGroup newGroupConf)
       throws RatisRequestFailedException {
     // notify the group leader of configuration change
     RaftClientReply reply;
-    try (RatisClient client = getRaftClient(newGroupConf)) {
-      reply =
-          client.getRaftClient().admin().setConfiguration(new ArrayList<>(newGroupConf.getPeers()));
-      if (!reply.isSuccess()) {
+    try (RatisClient client = getConfigurationRaftClient(newGroupConf)) {
+      while (true) {
```

Comment:
do we need this while(true) now? as we have already done endless retry in raftclient

Reply:
I think it's no problem,  but i can remove it